src/api/web3.js

```diff
@@ -72,18 +75,37 @@ async function getWeb3() {
       networkState.expectedNetworkName = getNetworkName(
         networkState.expectedNetworkId
       )
-
       if (window.ethereum) {
         web3 = new Web3(window.ethereum)
       } else if (window.web3 && window.web3.currentProvider) {
         web3 = new Web3(window.web3.currentProvider)
         networkState.readOnly = false
       } else {
-        console.log(
-          'No web3 instance injected. Falling back to cloud provider.'
-        )
-        web3 = new Web3(getNetworkProviderUrl(networkState.expectedNetworkId))
-        networkState.readOnly = true
+        //local node
+        try {
+          await fetch('http://localhost:8545')
+        } catch (error) {
+          if (
+            !(
+              error.readyState !== 4 &&
+              (error.status === 400 || error.status === 200)
+            )
+          ) {
+            localEndpoint = true
+            web3 = new Web3(new Web3.providers.HttpProvider(url))
+          } else {
+            web3 = new Web3(
+              getNetworkProviderUrl(networkState.expectedNetworkId)
+            )
+            networkState.readOnly = true
+          }
+        } finally {
+          if (web3) {
+            console.log('Success: Local node active')
+          } else {
+            console.log('Falling back to cloud provider.')
```

Comment:
Actually if web3 is not set then no connection was possible, and should show an error. Technically this `else` condition should never get executed.